.github/workflows/pre-release.yml

```diff
@@ -77,6 +77,95 @@ jobs:
           CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
           ASYNC_STD_THREAD_COUNT: 4
 
+  checks-ubuntu-24:
+    name: Code checks on Ubuntu 24.04
+    runs-on: ubuntu-latest
+    container: ubuntu:24.04
```

Comment:
To avoid all the code duplication, you can add a matrix over multiple container images. This way we can have less code and more flexibility. I suggest putting the following images:

- `ubuntu:24.04`
- `ubuntu:22.04`
- `ubuntu:20.04`

You can also experiment some Alpine, RedHat etc.

Reply:
The issue is that if we add a matrix it would mean that all runs will occur on containers, but we want the base workflow to run on baremetal, which is why I made a copy.

Reply:
[GitHub-hosted runners are VMs anyway.](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#overview-of-github-hosted-runners)

Reply:
And AFAICT, there is no additional virtualization imposed by Docker in this case, so the only real difference is the OS.